pkg/resources/task_acceptance_test.go

```diff
@@ -1,564 +1,970 @@
+				Check: assert.AssertThat(t,
+					resourceassert.TaskResource(t, firstTaskStandaloneModelDisabled.ResourceReference()).
+						HasScheduleString(schedule).
+						HasEnabledString(r.BooleanFalse).
+						HasSuspendTaskAfterNumFailuresString("10"),
+					resourceshowoutputassert.TaskShowOutput(t, firstTaskStandaloneModelDisabled.ResourceReference()).
+						HasSchedule(schedule).
+						HasState(sdk.TaskStateSuspended),
+					resourceassert.TaskResource(t, secondTaskStandaloneModelDisabled.ResourceReference()).
+						HasScheduleString(schedule).
+						HasEnabledString(r.BooleanFalse),
+					resourceshowoutputassert.TaskShowOutput(t, secondTaskStandaloneModelDisabled.ResourceReference()).
+						HasSchedule(schedule).
+						HasState(sdk.TaskStateSuspended),
+				),
+			},
+		},
+	})
 }
 
-`
-	return fmt.Sprintf(s, name, databaseName, schemaName, name, databaseName, schemaName)
-}
+func TestAcc_Task_ConvertStandaloneTaskToFinalizer(t *testing.T) {
+	_ = testenvs.GetOrSkipTest(t, testenvs.EnableAcceptance)
+	acc.TestAccPreCheck(t)
 
-func taskConfigManaged2(name, databaseName, schemaName, warehouseName string) string {
-	s := `
-resource "snowflake_warehouse" "wh" {
-	name = "%s-%s"
-}
+	rootTaskId := acc.TestClient().Ids.RandomSchemaObjectIdentifier()
+	finalizerTaskId := acc.TestClient().Ids.RandomSchemaObjectIdentifier()
+	statement := "SELECT 1"
+	schedule := "5 MINUTES"
 
-resource "snowflake_task" "managed_task" {
-	name     	  = "%s"
-	database  	  = "%s"
-	schema    	  = "%s"
-	sql_statement = "SELECT 1"
-	enabled  	  = true
-	schedule      = "5 MINUTE"
-	warehouse     = snowflake_warehouse.wh.name
-}
-`
-	return fmt.Sprintf(s, warehouseName, name, name, databaseName, schemaName)
-}
+	firstTaskStandaloneModel := model.TaskWithId("main_task", rootTaskId, true, statement).
+		WithSchedule(schedule).
+		WithSuspendTaskAfterNumFailures(1)
+	secondTaskStandaloneModel := model.TaskWithId("second_task", finalizerTaskId, true, statement).
+		WithSchedule(schedule)
 
-func taskConfigManaged3(name, databaseName, schemaName string) string {
-	s := `
-resource "snowflake_task" "managed_task" {
-	name     	                             = "%s"
-	database  	                             = "%s"
-	schema    	                             = "%s"
-	sql_statement                            = "SELECT 1"
-	enabled  	                             = true
-	schedule                                 = "5 MINUTE"
-    user_task_managed_initial_warehouse_size = "SMALL"
-}
-`
-	return fmt.Sprintf(s, name, databaseName, schemaName)
+	rootTaskModel := model.TaskWithId("main_task", rootTaskId, true, statement).
+		WithSchedule(schedule).
+		WithSuspendTaskAfterNumFailures(2)
+	childTaskModel := model.TaskWithId("second_task", finalizerTaskId, true, statement).
+		WithFinalize(rootTaskId.FullyQualifiedName())
+	childTaskModel.SetDependsOn([]string{rootTaskModel.ResourceReference()})
```

Comment:
I don't think DependsOn is needed here because we reference rootTask by finalizer field.

Reply:
But it's only referenced by id and not internally, meaning:
```terraform
finalize = "<identifier>"
```
is in config, not:
```terraform
finalize = snowflake_task.root_task.fully_qualified_name
```
That's why we need depends_on, or I misunderstood the comment?

Reply:
Yup, it's fine.